app/models/spotlight/reindexing_log_entry.rb

```diff
@@ -0,0 +1,42 @@
+module Spotlight
+  ##
+  # a log entry representing an attempt to reindex some number of records in an exhibit
+  class ReindexingLogEntry < ActiveRecord::Base
+    enum job_status: { unstarted: 0, in_progress: 1, succeeded: 2, failed: 3 }
+
+    belongs_to :exhibit, class_name: 'Spotlight::Exhibit'
+    belongs_to :user, class_name: '::User'
+
+    # null start times sort to the top, to more easily surface pending reindexing
+    default_scope { order('start_time IS NOT NULL, start_time DESC') }
+    scope :recent, -> { limit(5) }
+
+    def duration
+      end_time - start_time if end_time
+    end
+
+    def reindexing_started!
+      self.start_time = Time.zone.now
+      self.job_status = :in_progress
+      save
+    rescue
+      Rails.logger.error "unexpected error updating log entry to :in_progress from #{caller}"
+    end
+
+    def reindexing_succeeded!
+      self.end_time = Time.zone.now
+      self.job_status = :succeeded
```

Comment:
ok, got it, i see what you're saying.

yeah, so i did come across that functionality when reading up on enum, but didn't think to leverage it, maybe because it wouldn't set `end_time` by default, or maybe because i just hadn't used it previously.

would it be more conventionally rails-y to leave it as-is, or to override the `succeed!` method and have it do the time setting thing?  it does seem like both the status setting and start/end time setting should be encapsulated in a single method call.  is overriding something like `succeed!` and adding functionality encouraged or discouraged?